pkg/controller/network/expressroutecircuits/managed.go

```diff
@@ -0,0 +1,161 @@
+	"github.com/crossplane/provider-azure/pkg/clients/network"
+)
+
+// Error strings.
+const (
+	errNotExpressRouteCircuits    = "managed resource is not an ExpressRouteCircuits"
+	errCreateExpressRouteCircuits = "cannot create ExpressRouteCircuits"
+	errUpdateExpressRouteCircuits = "cannot update ExpressRouteCircuits"
+	errGetExpressRouteCircuits    = "cannot get ExpressRouteCircuits"
+	errDeleteExpressRouteCircuits = "cannot delete ExpressRouteCircuits"
+)
+
+// Setup adds a controller that reconciles ExpressRouteCircuits.
+func Setup(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(v1alpha3.ExpressRouteCircuitsGroupKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&v1alpha3.ExpressRouteCircuits{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha3.ExpressRouteCircuitsGroupVersionKind),
+			managed.WithConnectionPublishers(),
+			managed.WithExternalConnecter(&connecter{client: mgr.GetClient()}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connecter struct {
+	client client.Client
+}
+
+func (c *connecter) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	creds, auth, err := azureclients.GetAuthInfo(ctx, c.client, mg)
+	if err != nil {
+		return nil, err
+	}
+	cl := azurenetwork.NewExpressRouteCircuitsClient(creds[azureclients.CredentialsKeySubscriptionID])
+	cl.Authorizer = auth
+	return &external{client: cl}, nil
+}
+
+type external struct {
+	client networkapi.ExpressRouteCircuitsClientAPI
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	exp, ok := mg.(*v1alpha3.ExpressRouteCircuits)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotExpressRouteCircuits)
+	}
+
+	az, err := e.client.Get(ctx, exp.Spec.ResourceGroupName, meta.GetExternalName(exp))
+
+	if azureclients.IsNotFound(err) {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetExpressRouteCircuits)
+	}
+
+	network.UpdateExpressRouteCircuitStatusFromAzure(exp, az)
+	exp.SetConditions(xpv1.Available())
+
+	o := managed.ExternalObservation{
+		ResourceExists:    true,
+		ConnectionDetails: managed.ConnectionDetails{},
+	}
+
+	return o, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	exp, ok := mg.(*v1alpha3.ExpressRouteCircuits)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotExpressRouteCircuits)
+	}
+
+	exp.Status.SetConditions(xpv1.Creating())
```

Comment:
```suggestion
```
crossplane-runtime can now [handle setting "creating" and "deleting" conditions](https://github.com/crossplane/crossplane-runtime/pull/242).